src/client/mod.rs

```diff
@@ -290,464 +215,122 @@ impl Client {
         Ok(Client {
             flags: flags,
             cluster_id: cluster_id,
-            mmv_path: mmv_path,
-            wi: MMVWriterInfo::new()
+            mmv_path: mmv_path
         })
     }
+    
+    pub fn export(&self, metrics: &mut [&mut MMVWriter]) -> io::Result<()> {
```

Comment:
The mutable references live only within the block of `export`, so metrics can be freely mutated after the call. The tests and examples demonstrate this.

Also not sure why to limit the client to export only once? If the user wanted to add/remove metrics under the same name or cluster ID, it would be the only way to do it. Doesn't Speed also allow starting/stopping the mmap multiple times?